.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,100 @@
+# Modified from: https://github.com/oats-center/isoblue/blob/master/.github/workflows/build-and-push-services.yml
+name: Build and Push Pong Containers
+
+on:
```

Comment:
This is a good question. I copied the logic from my old grad school project. The idea then (and now) is as follows:

All proposed code should be built in the same way that the final code is built so that any discrepancies in the build environment between dev machines and the release built system are accounted for. In addition, it should be pushed to the registry it can be tested by pulling from the registry instead of building. This allows us to test proposed code on our twin (or any other suitable test system) by just pulling from the compose instead of pulling the source in rebuilding, and removes any potential issues with our local build system. Finally, in an emergency, the exhibit itself can use proposed code from any published commit while a hotfix is deployed to the main codebase.

`${TAG:-latest}` instructs the docker-compose file to use the latest tag unless a different one is specified in an environment variable. When developing, using `docker compose ... --build` essentially instructs docker to build instead of pulling, negating the issue during development

Note, I am changing `latest` to `next`. This is a minor semantic change, but arises due to some problems with with latest tag (discussed [here](https://www.docker.com/blog/docker-best-practices-using-tags-and-labels-to-manage-docker-image-sprawl/) and [here](https://www.howtogeek.com/devops/understanding-dockers-latest-tag/)). Essentially, the idea is that you should always be specifying a version when pulling, and if you really want `next` better communicates the 'bleeding edge' part of the last commit to main (not the latest release)

Reply:
@pdschmir this is a test notification

Reply:
TODO: Move above explanation into comment somewhere